src/bin/tikv-ctl.rs

```diff
@@ -337,26 +366,64 @@ fn dump_all_region_info(db: DB, skip_tombstone: bool) {
         .unwrap();
 }
 
-fn dump_range(db: DB, from: String, to: Option<String>, limit: Option<u64>, cf: &str) {
+fn parse_ts_key_from_key(encode_key: Vec<u8>) -> (u64, String) {
+    let item_key = Key::from_encoded(encode_key);
+    let ts = item_key.decode_ts().unwrap();
+    let item_key = item_key.truncate_ts().unwrap();
+    let key_vec = item_key.encoded();
+
+    let key = String::from_utf8(key_vec.clone()).unwrap();
+    // println!("ts:{:?},key_vec:{:?}",ts,key.clone());
+    (ts, key)
+}
+
+fn dump_range(db: DB,
+              from: String,
+              to: Option<String>,
+              limit: Option<u64>,
+              cf: &str,
+              start_ts: Option<u64>,
+              commit_ts: Option<u64>) {
     let from = unescape(&from);
     let to = to.map_or_else(|| vec![0xff], |s| unescape(&s));
     let limit = limit.unwrap_or(u64::MAX);
 
     if limit == 0 {
         return;
     }
-
     let mut cnt = 0;
     db.scan_cf(cf,
                  &from,
                  &to,
                  true,
                  &mut |k, v| {
-                     println!("key: {}, value len: {}", escape(k), v.len());
-                     println!("{}", escape(v));
-                     cnt += 1;
-                     Ok(cnt < limit)
-                 })
+            let mut right_key = true;
+            match cf {
+                CF_DEFAULT => {
+                    let (ts, _) = parse_ts_key_from_key(escape(k).into_bytes());
```

Comment:
It seems I need the method parse_ts_key_from_key in [issues_1284](https://github.com/pingcap/tikv/issues/1284) @disksing